compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -206,12 +206,35 @@ object QuoteMatcher {
         // Matches an open term and wraps it into a lambda that provides the free variables
         case Apply(TypeApply(Ident(_), List(TypeTree())), SeqLiteral(args, _) :: Nil)
             if pattern.symbol.eq(defn.QuotedRuntimePatterns_higherOrderHole) =>
+
+          /* Some of method symbols in arguments of higher-order term hole are eta-expanded.
+           * e.g.
+           * g: (Int) => Int
+           * => {
+           *   def $anonfun(y: Int): Int = g(y)
+           *   closure($anonfun)
+           * }
+           *
+           * f: (using Int) => Int
+           * => f(using x)
+           * This function restores the symbol of the original method from
+           * the eta-expanded function.
+           */
+          def getCapturedIdent(arg: Tree)(using Context): Ident =
+            arg match
+              case id: Ident => id
+              case Apply(term, _) => getCapturedIdent(term)
+              case Block((ddef: DefDef) :: _, _) => getCapturedIdent(ddef.rhs)
+              // How should I deal with it?
+              case y => ???
+
           val env = summon[Env]
-          val capturedArgs = args.map(_.symbol)
-          val captureEnv = env.filter((k, v) => !capturedArgs.contains(v))
+          val capturedIds = args.map(getCapturedIdent)
+          val capturedSymbols = capturedIds.map(_.symbol)
```

Comment:
`getCapturedIdent` could return the `Symbol` directly. This way, we can avoid this extra `map`.

Reply:
Here I intended to use `capturedIds`  as a parameter to `matchedOpen` (we get compiler errors if we use `args` instead).

Reply:
According to your other comment on i17105.check, it's likely I need to change this part anyway.